src/core/pagerankGraph.js

```diff
@@ -296,6 +315,85 @@ export class PagerankGraph {
     );
   }
 
+  /**
+   * Serialize this graph into a PagerankJSON object.
+   *
+   * Returns a plain JavaScript object.
+   *
+   * For space efficency, we store the node scores as an array of numbers in
+   * node-address-sorted order, and we store the edge weights as two arrays of
+   * numbers in edge-address-sorted-order.
+   */
+  toJSON(): PagerankGraphJSON {
+    this._verifyGraphNotModified();
+
+    const graphJSON = this.graph().toJSON();
+    const nodes = sortedNodeAddressesFromJSON(graphJSON);
+    const scores = new Array(nodes.length);
```

Comment:
should you add a type to `scores` eg like `toWeights` and `froWeights` below ?

Reply:
Also, out of curiosity, why the for-loops and not: 
```javascript
const scores = nodes.map((n) => NullUtil.get(this._scores.get(n))
```

Reply:
Thanks for the review! I'm not sure why I used the for loops, these maps are cleaner; I've modified the code. I also added the explicit type annotations per your suggestion.